src/azure-account.api.d.ts

```diff
@@ -4,10 +4,10 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Event, Terminal, Progress, CancellationToken } from 'vscode';
-import { ServiceClientCredentials } from 'ms-rest';
-import { AzureEnvironment } from 'ms-rest-azure';
-import { SubscriptionModels } from 'azure-arm-resource';
+import { ServiceClientCredentials } from '@azure/ms-rest-js';
+import { Environment } from '@azure/ms-rest-azure-env';
 import { ReadStream } from 'fs';
+import { SubscriptionModels } from '@azure/arm-subscriptions';
```

Comment:
I wonder if the types we actually use are compatible between new and old SDKs. If so, could we inline the interfaces here and not import any of the new (nor old) SDK's types?

Reply:
From what I can tell, they don't seem compatible. There are a couple of fields a few layers under these types that are incompatible. I remember one case had an issue with the way the signRequest method was defined (callback with void vs an awaitable Promise<T>). 

From what I understand there are small changes in the typings (callback -> thenable, string -> enum, etc.)

In this case, using the `SubscriptionModels` from `azure-arm-resource` leads to an error like: 
```
 Types of property 'subscription' are incompatible.
      Type 'import("c:/repo/vscode-azure-account/node_modules/azure-arm-resource/lib/subscription/models/index").Subscription' is not assignable to type 'import("c:/repo/vscode-azure-account/node_modules/@azure/arm-subscriptions/esm/models/index").Subscription'.
        Types of property 'state' are incompatible.
          Type 'string | undefined' is not assignable to type '"Enabled" | "Warned" | "PastDue" | "Disabled" | "Deleted" | undefined'.
            Type 'string' is not assignable to type '"Enabled" | "Warned" | "PastDue" | "Disabled" | "Deleted" | undefined'.ts(2322)
```

Reply:
`string | undefined` is not assignable to `"Enabled" | "Warned" | "PastDue" | "Disabled" | "Deleted" | undefined`, but it _will_ work the other way around.